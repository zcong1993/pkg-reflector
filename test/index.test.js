const path = require('path')
const spawn = require('cross-spawn')

const testRoot = __dirname

const prf = path.resolve(__dirname, '../', 'bin', 'pkg-reflector')

test('default usage', () => {
  const testDir = path.resolve(__dirname, 'test1')
  const testJson = path.resolve(testDir, 'package.json')
  spawn.sync('cp', ['-a', './fixture', './test1'], { cwd: testRoot })
  spawn.sync(prf, ['./*'], { cwd: testDir })
  const pkg = require(testJson)
  expect(pkg).toMatchSnapshot()
  spawn.sync('rm', ['-rf', './test1'], { cwd: testRoot })
})

test('save dev', () => {
  const testDir = path.resolve(__dirname, 'test2')
  const testJson = path.resolve(testDir, 'package.json')
  spawn.sync('cp', ['-a', './fixture', './test2'], { cwd: testRoot })
  spawn.sync(prf, ['./*', '-d'], { cwd: testDir })
  const pkg = require(testJson)
  expect(pkg).toMatchSnapshot()
  spawn.sync('rm', ['-rf', './test2'], { cwd: testRoot })
})
