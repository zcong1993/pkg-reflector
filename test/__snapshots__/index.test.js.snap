// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default usage 1`] = `
Object {
  "dependencies": Object {
    "lodash": "^4.17.4",
    "vue": "^2.4.2",
  },
}
`;

exports[`save dev 1`] = `
Object {
  "devDependencies": Object {
    "lodash": "^4.17.4",
    "vue": "^2.4.2",
  },
}
`;
